src/System.Private.ServiceModel/src/Internals/System/Runtime/MruCache.cs

```diff
@@ -202,7 +209,12 @@ public bool TryGetValue(TKey key, out TValue value)
 
         public void Dispose()
         {
-            Dispose(true);
+            int refCount = Interlocked.Decrement(ref _refCount);
```

Comment:
Isn't there a race with this ref counting? Thread A can begin disposing and decrement, Thread B can AddRef and increment, and then Thread B can continue to dispose, even though Thread B thinks they have a valid reference.

Reply:
Although it won't be used in that way as this is only possible to hit when a single instance of HttpTransportBindingElement is reused and it will be reused sequentially, I agree that having this have potential race conditions if that usage changes isn't the best idea. The only way I can see to make this bullet proof is with a lock. Do the ref counting in the lock and change AddRef to return a bool if it can't obtain increment it because it is disposed.